Add extra metadata (as comments) to Inductor generated code

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96690
* __->__ #96581
* #96565

New output
<img width="942" alt="image" src="https://user-images.githubusercontent.com/6355099/224794006-a993a2a8-d6ff-49da-8891-7b2373030a3d.png">


cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

Chillee committed:
Add extra metadata (as comments) to Inductor generated code

[ghstack-poisoned]

Chillee committed:
Update on "Add extra metadata (as comments) to Inductor generated code"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Chillee committed:
Update on "Add extra metadata (as comments) to Inductor generated code"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

ezyang commented:
i'll let the flop counter reviewers handle this, but holler if you still need a review

ngimel reviewed (approved):
Cool, an example generated comment in PR description would be helpful

Chillee committed:
Update on "Add extra metadata (as comments) to Inductor generated code"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Chillee commented on torch/_inductor/triton_ops/autotune.py:
```diff
@@ -233,8 +233,8 @@ def start_graph():
         return
-    overall_time = sum(call[1] for call in collected_calls)
-    overall_gb = sum(call[2] for call in collected_calls)
+    overall_time = sum(call[0] for call in collected_calls)
```
It was a bug I fixed.

Chillee committed:
Update on "Add extra metadata (as comments) to Inductor generated code"


New output
<img width="942" alt="image" src="https://user-images.githubusercontent.com/6355099/224794006-a993a2a8-d6ff-49da-8891-7b2373030a3d.png">


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

shunting314 commented on torch/_inductor/utils.py:
```diff
@@ -254,13 +255,39 @@ def get_fused_kernel_name(node_schedule):
+    for node in inductor_nodes:
+        if "original_aten" in node.meta:
+            original_aten_dict[
+                str(node.meta["original_aten"]._overloadpacket)
```
Is this `original_aten` meta data initially being set by `make_fx` or somewhere else?

Chillee replied:
Yeah https://github.com/pytorch/pytorch/blob/master/torch/fx/experimental/proxy_tensor.py#L477

Chillee committed:
Update on "Add extra metadata (as comments) to Inductor generated code"


New output
<img width="942" alt="image" src="https://user-images.githubusercontent.com/6355099/224794006-a993a2a8-d6ff-49da-8891-7b2373030a3d.png">


cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Chillee commented:
@pytorchbot merge

Chillee commented:
@pytorchbot merge
